kernel/boot_info/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+//! This crate provides an abstraction over multiboot2 and UEFI boot
+//! information.
+//!
+//! It allows the kernel's initialisation to be the same, regardless of whether
+//! it was booted using BIOS or UEFI.
+
+#![feature(type_alias_impl_trait)]
+#![no_std]
+
+pub mod multiboot2;
+
+use core::{iter::Iterator, ops::Range};
+use memory_structs::{PhysicalAddress, VirtualAddress};
+
+pub trait MemoryRegion {
+    /// Returns the region's starting physical address.
+    fn start(&self) -> PhysicalAddress;
+
+    /// Returns the region's length.
+    fn len(&self) -> usize;
+
+    /// Returns whether the region can be used by the frame allocator.
+    fn is_usable(&self) -> bool;
```

Comment:
"usable" here means that it's available for general-purpose usage is not reserved for any other specific use, right? 

I'm fine with a bool as long as it's sufficient for our purpose, but I think we should add a bit more detail about what "usable" really means, and what it means if it's not usable. Perhaps an enum (non-exhaustive?) would be better, and then we could more easily add different types if we need to disambiguate in the future. 

Open to hearing your thoughts here.

Reply:
I initially had an enum, but the memory startup code only checks whether or not the memory is usable, so I figured it would be easier to use a boolean.